tests/unit/unit_test_main.cpp

```diff
@@ -9,5 +9,17 @@
 // terms of the GNU Lesser General Public License (as published by the Free
 // Software Foundation) version 2.1 dated February 1999.
 
-#define CATCH_CONFIG_MAIN     // This tells Catch to provide a main() - only do this in one cpp file
+#define CATCH_CONFIG_RUNNER
 #include "catch.hpp"
+#include "mfem.hpp"
+
+int main(int argc, char *argv[])
+{
+#ifdef MFEM_USE_MPI
+   mfem::MPI_Session mpi;
+#endif
```

Comment:
Note that some of the tests (e.g. the data collection tests that do file I/O) will fail if you run in parallel. Maybe we should force only one MPI rank?

Reply:
I think this is the only thing still outstanding from my review. Ideas?

Reply:
I missed that. If those fail, my suggestion is to not run them in parallel if they are serial only. E.g.

if compiled with MPI -> if num_ranks > 1 -> skip this test by returning a success

Reply:
If I understand correctly, this is mostly about compilation -- the unit tests are actually run without `mpirun` so there shouldn't be any issues with the current tests.